app/services/classification/base.rb

```diff
@@ -35,9 +35,9 @@ def possible_value_orders
 
     def values_hash(options = {})
       values = Hash.new { |h,k| h[k] = {} }
-      all_lookup_values = LookupValue.where(:match => path2matches).where(:lookup_key_id => class_parameters)
+      all_lookup_values = LookupValue.where(:match => path2matches).where(:lookup_key_id => class_parameters, :use_puppet_default => false).includes(:lookup_key).to_a
```

Comment:
instead of doing a query for ever class param, let's fetch all of them and just select the relevant ones from the array.